docs/transaction-verification.md

```diff
@@ -0,0 +1,164 @@
+Transfers where a user wants to send CCD from one account to another is an example hereof. 
+ 
+These types of transactions are all subject to a verification process. Naturally the verification process is different 
+for each of the types and thus we will now describe what is checked for each type of transaction.
+However there is some common verification that is being carried out before the transaction is deserialized into one of the above mentioned types.
+ 
+## Verification
+The verification of transactions takes place at two times in the lifetime of a transaction being sent to the node.
+The first occurence of verification is when it is initially received by the node.
+The second time a transaction is being verified is just before the execution of it in the `scheduler`.
+To prevent duplicate work between the `initial` verification and the `pre execution` then the verification result from the 
+`initial` verification is stored alongside the transaction in the `transaction table`. The `scheduler` leverages it's verification process 
+by using the existing verification result when possible. This is especially to avoid duplicating the expensive checks such as signature verification.
+Ephemeral properties of transactions, such as `expiry`, balances of accounts etc. are always checked by the `scheduler` before executing the transaction.
+
+Transactions that are deemed invalid will per default not be relayed to the network. However the node can be configured in such a way that it will propagate messages to the network no matter the result of the verification process.
+ 
+### Initial verification 
+ 
+The verification process is dependent on the `context` of the transaction. 
+As stated above, the node can receive transactions individually or via a block. 
+Transactions received as part of a block will be verified in the context of the (alive) parent block if such one exist, 
+otherwise verification will be carried out in the context of the `last finalized block`.
+Individually received transactions are always verified in the context of the last finalized block.
+The reason for this discrepancy is that blocks may arrive out of order - this happens especially when the node is catching up on the chain, 
+and as such this heuristic aims to be the 'best guess' for which the transaction should be valid.
+ 
+If the transaction was received individually then it must meet all the criterias stated underneath in order to be taken into consideration for `consensus` 
+otherwise it will be rejected.
+However if the transaction was received as part of a block then the criterias are 'loosened up' a bit. 
+The reason is, when the node is not up to date with the chain then there is a probability that it just might not know about the the actual state of the chain and 
+as such the transaction could've become valid further on the chain.
```

Comment:
I think another thing that is worth pointing out is that without just executing the block, it is impossible to say which transactions are valid or not in some cases. Since validity of a transaction in general depends on all previous transactions.